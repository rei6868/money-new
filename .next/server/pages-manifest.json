{"/_app":"pages/_app.js","/accounts":"pages/accounts.html","/api/accounts/[id]":"pages/api/accounts/[id].js","/api/accounts":"pages/api/accounts.js","/api/accounts/types":"pages/api/accounts/types.js","/api/categories/[id]":"pages/api/categories/[id].js","/api/people/[id]":"pages/api/people/[id].js","/api/categories":"pages/api/categories.js","/api/people":"pages/api/people.js","/api/people/list":"pages/api/people/list.js","/api/shops/[id]":"pages/api/shops/[id].js","/api/shops":"pages/api/shops.js","/api/transactions/action":"pages/api/transactions/action.js","/api/transactions/[id]":"pages/api/transactions/[id].js","/api/transactions/columns":"pages/api/transactions/columns.js","/api/transactions":"pages/api/transactions.js","/api/transactions/meta":"pages/api/transactions/meta.js","/api/transactions/restore":"pages/api/transactions/restore.js","/api/transactions/types":"pages/api/transactions/types.js","/api/transactions/selection":"pages/api/transactions/selection.js","/attendance":"pages/attendance.html","/batch-input":"pages/batch-input.html","/cashback/flow":"pages/cashback/flow.html","/cashback/ledger":"pages/cashback/ledger.html","/cashback/movements":"pages/cashback/movements.html","/cashback/summary":"pages/cashback/summary.html","/category":"pages/category.html","/dashboard":"pages/dashboard.html","/debt":"pages/debt.html","/debt/list":"pages/debt/list.html","/debt/movements":"pages/debt/movements.html","/geo-information":"pages/geo-information.html","/hub":"pages/hub.html","/_error":"pages/_error.js","/":"pages/index.html","/inventory":"pages/inventory.html","/invoice":"pages/invoice.html","/login":"pages/login.html","/orders":"pages/orders.html","/overview":"pages/overview.html","/people":"pages/people.html","/product":"pages/product.html","/quick-add/settings":"pages/quick-add/settings.html","/reports":"pages/reports.html","/settings":"pages/settings.html","/shop":"pages/shop.html","/subscription":"pages/subscription.html","/transactions":"pages/transactions.html","/users":"pages/users.html","/_document":"pages/_document.js","/404":"pages/404.html"}