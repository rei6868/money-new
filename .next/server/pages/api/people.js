"use strict";(()=>{var e={};e.id=4775,e.ids=[4775],e.modules={1081:e=>{e.exports=require("dotenv/config")},145:e=>{e.exports=require("next/dist/compiled/next-server/pages-api.runtime.prod.js")},7612:e=>{e.exports=import("drizzle-orm")},1749:e=>{e.exports=import("drizzle-orm/node-postgres")},8358:e=>{e.exports=import("drizzle-orm/pg-core")},8678:e=>{e.exports=import("pg")},6113:e=>{e.exports=require("crypto")},76:(e,t,r)=>{r.a(e,async(e,o)=>{try{r.r(t),r.d(t,{config:()=>d,default:()=>i,routeModule:()=>c});var a=r(1802),n=r(7153),s=r(6249),l=r(1429),u=e([l]);l=(u.then?(await u)():u)[0];let i=(0,s.l)(l,"default"),d=(0,s.l)(l,"config"),c=new a.PagesAPIRouteModule({definition:{kind:n.x.PAGES_API,page:"/api/people",pathname:"/api/people",bundlePath:"",filename:""},userland:l});o()}catch(e){o(e)}})},1429:(e,t,r)=>{r.a(e,async(e,o)=>{try{r.r(t),r.d(t,{default:()=>handler});var a=r(7612),n=r(6113),s=r(4379),l=r(6340),u=e([a,s,l]);[a,s,l]=u.then?(await u)():u;let i=[{personId:"mock-person-1",fullName:"John Doe",contactInfo:"john@example.com",status:"active",groupId:null,imgUrl:null,note:"Primary user",createdAt:new Date().toISOString(),updatedAt:new Date().toISOString()},{personId:"mock-person-2",fullName:"Jane Smith",contactInfo:"jane@example.com",status:"active",groupId:null,imgUrl:null,note:"Secondary user",createdAt:new Date().toISOString(),updatedAt:new Date().toISOString()}];async function handler(e,t){let r=s.db;if(!r){if(console.warn("Database connection is not configured - using mock data"),"GET"===e.method){t.status(200).json(i);return}if("POST"===e.method){let r=e.body||{};if(!r.fullName||!r.status){t.status(400).json({error:"Validation failed",details:"fullName and status are required"});return}let o={personId:`mock-person-${Date.now()}`,fullName:r.fullName,contactInfo:r.contactInfo||null,status:r.status,groupId:r.groupId||null,imgUrl:r.imgUrl||null,note:r.note||null,createdAt:new Date().toISOString(),updatedAt:new Date().toISOString()};i.push(o),t.status(201).json(o);return}t.status(405).json({error:"Method not allowed"});return}if("GET"===e.method){try{let e=await r.select().from(l.M).orderBy((0,a.asc)(l.M.fullName));t.status(200).json(e)}catch(r){console.error("Failed to fetch people",r);let e=r instanceof Error?r.message:"Unknown error";t.status(500).json({error:e})}return}if("POST"===e.method){try{let o=e.body||{};if(!o.fullName||!o.status){t.status(400).json({error:"Validation failed",details:"fullName and status are required"});return}let a={personId:(0,n.randomUUID)(),fullName:o.fullName,status:o.status,contactInfo:o.contactInfo||null,groupId:o.groupId||null,imgUrl:o.imgUrl||null,note:o.note||null},s=await r.insert(l.M).values(a).returning();t.status(201).json(s[0])}catch(r){console.error("Failed to create person",r);let e=r instanceof Error?r.message:"Unknown error";t.status(500).json({error:e})}return}t.setHeader("Allow",["GET","POST"]),t.status(405).json({error:"Method not allowed"})}o()}catch(e){o(e)}})}};var t=require("../../webpack-api-runtime.js");t.C(e);var __webpack_exec__=e=>t(t.s=e),r=t.X(0,[4222,4379,6340],()=>__webpack_exec__(76));module.exports=r})();