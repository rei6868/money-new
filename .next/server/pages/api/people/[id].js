"use strict";(()=>{var e={};e.id=44,e.ids=[44],e.modules={1081:e=>{e.exports=require("dotenv/config")},145:e=>{e.exports=require("next/dist/compiled/next-server/pages-api.runtime.prod.js")},7612:e=>{e.exports=import("drizzle-orm")},1749:e=>{e.exports=import("drizzle-orm/node-postgres")},8358:e=>{e.exports=import("drizzle-orm/pg-core")},8678:e=>{e.exports=import("pg")},791:(e,t,r)=>{r.a(e,async(e,o)=>{try{r.r(t),r.d(t,{config:()=>l,default:()=>u,routeModule:()=>c});var n=r(1802),s=r(7153),a=r(6249),i=r(2884),d=e([i]);i=(d.then?(await d)():d)[0];let u=(0,a.l)(i,"default"),l=(0,a.l)(i,"config"),c=new n.PagesAPIRouteModule({definition:{kind:s.x.PAGES_API,page:"/api/people/[id]",pathname:"/api/people/[id]",bundlePath:"",filename:""},userland:i});o()}catch(e){o(e)}})},2884:(e,t,r)=>{r.a(e,async(e,o)=>{try{r.r(t),r.d(t,{default:()=>handler});var n=r(7612),s=r(4379),a=r(6340),i=e([n,s,a]);async function handler(e,t){let r=s.db;if(!r){console.error("Database connection is not configured"),t.status(500).json({error:"Database connection is not configured"});return}let{id:o}=e.query,i=Array.isArray(o)?o[0]:o;if(!i){t.status(400).json({message:"Person id is required"});return}if("GET"===e.method){try{let e=await r.select().from(a.M).where((0,n.eq)(a.M.personId,i));if(0===e.length){t.status(404).json({message:"Person not found"});return}t.status(200).json(e[0])}catch(r){console.error(`Failed to fetch person with id ${i}`,r);let e=r instanceof Error?r.message:"Unknown error";t.status(500).json({error:e})}return}if("PUT"===e.method||"PATCH"===e.method){try{let o=e.body||{},s={};if(void 0!==o.fullName&&(s.fullName=o.fullName),void 0!==o.status&&(s.status=o.status),void 0!==o.contactInfo&&(s.contactInfo=o.contactInfo),void 0!==o.groupId&&(s.groupId=o.groupId),void 0!==o.imgUrl&&(s.imgUrl=o.imgUrl),void 0!==o.note&&(s.note=o.note),s.updatedAt=new Date,0===Object.keys(s).length){t.status(400).json({error:"No valid fields to update"});return}let d=await r.update(a.M).set(s).where((0,n.eq)(a.M.personId,i)).returning();if(0===d.length){t.status(404).json({message:"Person not found"});return}t.status(200).json(d[0])}catch(r){console.error(`Failed to update person with id ${i}`,r);let e=r instanceof Error?r.message:"Unknown error";t.status(500).json({error:e})}return}if("DELETE"===e.method){try{let e=await r.delete(a.M).where((0,n.eq)(a.M.personId,i)).returning();if(0===e.length){t.status(404).json({message:"Person not found"});return}t.status(200).json(e[0])}catch(r){console.error(`Failed to delete person with id ${i}`,r);let e=r instanceof Error?r.message:"Unknown error";t.status(500).json({error:e})}return}t.setHeader("Allow",["GET","PUT","PATCH","DELETE"]),t.status(405).json({error:"Method not allowed"})}[n,s,a]=i.then?(await i)():i,o()}catch(e){o(e)}})}};var t=require("../../../webpack-api-runtime.js");t.C(e);var __webpack_exec__=e=>t(t.s=e),r=t.X(0,[4222,4379,6340],()=>__webpack_exec__(791));module.exports=r})();