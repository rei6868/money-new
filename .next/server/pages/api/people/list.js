"use strict";(()=>{var e={};e.id=8375,e.ids=[8375],e.modules={1081:e=>{e.exports=require("dotenv/config")},145:e=>{e.exports=require("next/dist/compiled/next-server/pages-api.runtime.prod.js")},7612:e=>{e.exports=import("drizzle-orm")},1749:e=>{e.exports=import("drizzle-orm/node-postgres")},8358:e=>{e.exports=import("drizzle-orm/pg-core")},8678:e=>{e.exports=import("pg")},7081:(e,r,t)=>{t.a(e,async(e,o)=>{try{t.r(r),t.d(r,{config:()=>c,default:()=>d,routeModule:()=>p});var a=t(1802),s=t(7153),n=t(6249),i=t(8850),l=e([i]);i=(l.then?(await l)():l)[0];let d=(0,n.l)(i,"default"),c=(0,n.l)(i,"config"),p=new a.PagesAPIRouteModule({definition:{kind:s.x.PAGES_API,page:"/api/people/list",pathname:"/api/people/list",bundlePath:"",filename:""},userland:i});o()}catch(e){o(e)}})},8850:(e,r,t)=>{t.a(e,async(e,o)=>{try{t.r(r),t.d(r,{default:()=>handler});var a=t(7612),s=t(4379),n=t(6340),i=e([a,s,n]);async function handler(e,r){let t=s.db;if(!t){console.error("Database connection is not configured"),r.status(500).json({error:"Database connection is not configured"});return}if("GET"===e.method){try{let e=await t.select({personId:n.M.personId,fullName:n.M.fullName}).from(n.M).where((0,a.eq)(n.M.status,"active")).orderBy((0,a.asc)(n.M.fullName));r.status(200).json(e)}catch(t){console.error("Failed to fetch people list",t);let e=t instanceof Error?t.message:"Unknown error";r.status(500).json({error:e})}return}r.setHeader("Allow",["GET"]),r.status(405).json({error:"Method not allowed"})}[a,s,n]=i.then?(await i)():i,o()}catch(e){o(e)}})}};var r=require("../../../webpack-api-runtime.js");r.C(e);var __webpack_exec__=e=>r(r.s=e),t=r.X(0,[4222,4379,6340],()=>__webpack_exec__(7081));module.exports=t})();