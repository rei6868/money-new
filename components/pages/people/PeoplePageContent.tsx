import { useMemo } from 'react';

import PageSection from '../../layout/page/PageSection';
import pageSectionStyles from '../../layout/page/PageSection.module.css';

export type PeopleRecord = {
  personId: string;
  fullName: string;
  status: string;
};

export type PeoplePageContentProps = {
  people: PeopleRecord[];
  isLoading: boolean;
  error: string | null;
  onAddNew?: () => void;
  onEditPerson?: (person: PeopleRecord) => void;
  onDeletePerson?: (person: PeopleRecord) => void;
};

export function PeoplePageContent({
  people,
  isLoading,
  error,
  onAddNew,
  onEditPerson,
  onDeletePerson,
}: PeoplePageContentProps) {
  const rows = useMemo(() => people ?? [], [people]);
  const hasData = rows.length > 0;

  return (
    <PageSection
      title="People"
      description="Manage individuals linked to accounts, transactions, and reimbursements."
      actions={
        <div className={pageSectionStyles.buttonRow}>
          <button
            type="button"
            onClick={() => onAddNew?.()}
            className={pageSectionStyles.primaryButton}
          >
            Add New Person
          </button>
        </div>
      }
    >
      {isLoading ? <p className={pageSectionStyles.messageText}>Loading peopleâ€¦</p> : null}
      {!isLoading && error ? (
        <p className={pageSectionStyles.errorText} role="alert">
          {error}
        </p>
      ) : null}
      {!isLoading && !error ? (
        <div className={pageSectionStyles.tableWrapper}>
          <table className={pageSectionStyles.table}>
            <thead>
              <tr>
                <th scope="col" className={pageSectionStyles.tableCell}>
                  Full Name
                </th>
                <th scope="col" className={pageSectionStyles.tableCell}>
                  Status
                </th>
                <th scope="col" className={pageSectionStyles.tableCell}>
                  Actions
                </th>
              </tr>
            </thead>
            <tbody>
              {hasData ? (
                rows.map((person) => (
                  <tr key={person.personId}>
                    <td className={pageSectionStyles.tableCell}>{person.fullName}</td>
                    <td className={`${pageSectionStyles.tableCell} ${pageSectionStyles.capitalize}`}>
                      {person.status}
                    </td>
                    <td className={pageSectionStyles.tableCellActions}>
                      <div className={pageSectionStyles.tableRowActions}>
                        <button
                          type="button"
                          onClick={() => onEditPerson?.(person)}
                          className={pageSectionStyles.secondaryButton}
                        >
                          Edit
                        </button>
                        <button
                          type="button"
                          onClick={() => onDeletePerson?.(person)}
                          className={pageSectionStyles.dangerButton}
                        >
                          Delete
                        </button>
                      </div>
                    </td>
                  </tr>
                ))
              ) : (
                <tr>
                  <td colSpan={3} className={pageSectionStyles.tableCell}>
                    <div className={pageSectionStyles.emptyState}>No people found yet.</div>
                  </td>
                </tr>
              )}
            </tbody>
          </table>
        </div>
      ) : null}
    </PageSection>
  );
}

export default PeoplePageContent;
