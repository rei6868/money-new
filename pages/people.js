import { useEffect, useMemo, useState } from 'react';
import { useRouter } from 'next/router';

import AppLayout from '../components/AppLayout';
import { useRequireAuth } from '../hooks/useRequireAuth';

export default function PeoplePage() {
  const router = useRouter();
  const { isAuthenticated, isLoading: authLoading } = useRequireAuth();
  const [people, setPeople] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(null);

  useEffect(() => {
    if (authLoading || !isAuthenticated) {
      return;
    }

    let cancelled = false;

    const fetchPeople = async () => {
      setIsLoading(true);
      setError(null);

      try {
        const response = await fetch('/api/people');
        if (!response.ok) {
          throw new Error(`Request failed with status ${response.status}`);
        }

        const payload = await response.json();
        const peopleList = Array.isArray(payload) ? payload : payload?.people;
        if (!cancelled) {
          setPeople(Array.isArray(peopleList) ? peopleList : []);
          if (!Array.isArray(peopleList)) {
            console.warn('Unexpected response shape when fetching people', payload);
          }
        }
      } catch (err) {
        if (!cancelled) {
          console.error('Failed to load people', err);
          setError('Unable to load people right now.');
        }
      } finally {
        if (!cancelled) {
          setIsLoading(false);
        }
      }
    };

    fetchPeople();

    return () => {
      cancelled = true;
    };
  }, [authLoading, isAuthenticated]);

  const handleAddNew = () => {
    router.push('/people/new').catch((err) => {
      console.error('Failed to navigate to create person page', err);
    });
  };

  const rows = useMemo(() => people ?? [], [people]);

  if (authLoading || !isAuthenticated) {
    return null;
  }

  return (
    <AppLayout
      title="People"
      subtitle="Manage individuals linked to accounts, transactions, and reimbursements."
    >
      <div style={{ display: 'flex', justifyContent: 'space-between', marginBottom: '1rem' }}>
        <button type="button" onClick={handleAddNew} style={{ padding: '0.5rem 1rem', cursor: 'pointer' }}>
          Add New Person
        </button>
      </div>

      {isLoading && <p>Loading peopleâ€¦</p>}
      {error && !isLoading && <p style={{ color: 'red' }}>{error}</p>}

      {!isLoading && !error && (
        <table style={{ width: '100%', borderCollapse: 'collapse' }}>
          <thead>
            <tr>
              <th style={{ textAlign: 'left', borderBottom: '1px solid #ccc', padding: '0.5rem' }}>Full Name</th>
              <th style={{ textAlign: 'left', borderBottom: '1px solid #ccc', padding: '0.5rem' }}>Status</th>
              <th style={{ textAlign: 'left', borderBottom: '1px solid #ccc', padding: '0.5rem' }}>Actions</th>
            </tr>
          </thead>
          <tbody>
            {rows.length === 0 ? (
              <tr>
                <td colSpan={3} style={{ padding: '0.75rem' }}>
                  No people found yet.
                </td>
              </tr>
            ) : (
              rows.map((person) => (
                <tr key={person.personId} style={{ borderBottom: '1px solid #eee' }}>
                  <td style={{ padding: '0.75rem' }}>{person.fullName}</td>
                  <td style={{ padding: '0.75rem', textTransform: 'capitalize' }}>{person.status}</td>
                  <td style={{ padding: '0.75rem' }}>
                    <button
                      type="button"
                      onClick={() => console.log('Edit person', person.personId)}
                      style={{ marginRight: '0.5rem', padding: '0.25rem 0.5rem', cursor: 'pointer' }}
                    >
                      Edit
                    </button>
                    <button
                      type="button"
                      onClick={() => console.log('Delete person', person.personId)}
                      style={{ padding: '0.25rem 0.5rem', cursor: 'pointer' }}
                    >
                      Delete
                    </button>
                  </td>
                </tr>
              ))
            )}
          </tbody>
        </table>
      )}
    </AppLayout>
  );
}
