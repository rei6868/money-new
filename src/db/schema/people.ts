import { pgEnum, pgTable, text, timestamp, varchar } from "drizzle-orm/pg-core";

/**
 * Enumerates the lifecycle state for each person record within the platform.
 * This ensures downstream services treat active vs. inactive vs. archived
 * people consistently.
 */
export const personStatusEnum = pgEnum("person_status", [
  "active",
  "inactive",
  "archived",
]);

/**
 * People table captures all debtors and financial contacts that participate in
 * the platform, independent of user accounts. Each column is documented for
 * clarity across engineering and product stakeholders.
 */
export const people = pgTable("people", {
  /**
   * Primary key for the person. Uses string identifiers (e.g. UUID) to support
   * integration with external systems or imports.
   */
  personId: varchar("person_id", { length: 36 }).primaryKey(),

  /**
   * Full name displayed across UI surfaces and reports. Required so each record
   * is human readable and easy to identify by operations teams.
   */
  fullName: varchar("full_name", { length: 180 }).notNull(),

  /**
   * Optional contact information (phone number, email, messenger handle). This
   * field is stored as free-form text to accommodate multiple formats.
   */
  contactInfo: text("contact_info"),

  /**
   * Lifecycle status dictating whether the person is actively associated with
   * current transactions, paused, or archived from everyday views.
   */
  status: personStatusEnum("status").notNull(),

  /**
   * Optional reference to a group or household identifier. Enables future
   * linkage once group management tables are introduced.
   */
  groupId: varchar("group_id", { length: 36 }),

  /**
   * Optional URL to the person's avatar stored on a CDN such as Cloudinary.
   * Used on dashboards, cards, and mobile views for visual recognition.
   */
  imgUrl: text("img_url"),

  /**
   * Free-form notes for operations teams (e.g. debtor remarks or preferred
   * contact methods). Optional to avoid clutter when not needed.
   */
  note: text("note"),

  /**
   * Creation timestamp. Defaults to the database clock so record insertion time
   * is always captured for auditing.
   */
  createdAt: timestamp("created_at", { withTimezone: true }).defaultNow().notNull(),

  /**
   * Timestamp of the most recent update, enabling change tracking and audit
   * trails throughout the person's lifecycle.
   */
  updatedAt: timestamp("updated_at", { withTimezone: true }).defaultNow().notNull(),
});

export type Person = typeof people.$inferSelect;
export type NewPerson = typeof people.$inferInsert;
