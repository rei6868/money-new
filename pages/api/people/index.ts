import type { NextApiRequest, NextApiResponse } from "next";
import { asc } from "drizzle-orm";

import { db } from "../../../lib/db/client";
import { people, type NewPerson } from "../../../src/db/schema/people";

export default async function handler(req: NextApiRequest, res: NextApiResponse) {
  const database = db;

  if (!database) {
    console.error("Database connection is not configured");
    res.status(500).json({ error: "Database connection is not configured" });
    return;
  }

  if (req.method === "GET") {
    try {
      const result = await database.select().from(people).orderBy(asc(people.fullName));
      res.status(200).json(result);
    } catch (error) {
      console.error("Failed to fetch people", error);
      const message = error instanceof Error ? error.message : "Unknown error";
      res.status(500).json({ error: message });
    }
    return;
  }

  if (req.method === "POST") {
    try {
      const payload = req.body as NewPerson;
      const created = await database.insert(people).values(payload).returning();
      res.status(201).json(created[0]);
    } catch (error) {
      console.error("Failed to create person", error);
      const message = error instanceof Error ? error.message : "Unknown error";
      res.status(500).json({ error: message });
    }
    return;
  }

  res.setHeader("Allow", ["GET", "POST"]);
  res.status(405).json({ error: "Method not allowed" });
}
