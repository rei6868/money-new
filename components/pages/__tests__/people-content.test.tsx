import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

import { PeoplePageContent } from '../people/PeoplePageContent';

describe('PeoplePageContent', () => {
  const noop = () => {};

  it('renders the empty state consistently', () => {
    const html = renderToStaticMarkup(
      <PeoplePageContent
        people={[]}
        isLoading={false}
        error={null}
        onAddNew={noop}
        onEditPerson={noop}
        onDeletePerson={noop}
      />,
    );

    expect(html).toMatchSnapshot();
  });

  it('renders a table of people consistently', () => {
    const html = renderToStaticMarkup(
      <PeoplePageContent
        people={[
          { personId: '1', fullName: 'Alex Murphy', status: 'active' },
          { personId: '2', fullName: 'Sam Carter', status: 'inactive' },
        ]}
        isLoading={false}
        error={null}
        onAddNew={noop}
        onEditPerson={noop}
        onDeletePerson={noop}
      />,
    );

    expect(html).toMatchSnapshot();
  });
});
