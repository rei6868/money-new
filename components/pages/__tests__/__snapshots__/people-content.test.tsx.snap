// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeoplePageContent renders a table of people consistently 1`] = `"<section class="section"><div class="header"><div class="headerRow"><div class="heading"><h2 class="title">People</h2><p class="subtitle">Manage individuals linked to accounts, transactions, and reimbursements.</p></div><div class="actions"><div class="buttonRow"><button type="button" class="primaryButton">Add New Person</button></div></div></div></div><div class="body"><div class="tableWrapper"><table class="table"><thead><tr><th scope="col" class="tableCell">Full Name</th><th scope="col" class="tableCell">Status</th><th scope="col" class="tableCell">Actions</th></tr></thead><tbody><tr><td class="tableCell">Alex Murphy</td><td class="tableCell capitalize">active</td><td class="tableCellActions"><div class="tableRowActions"><button type="button" class="secondaryButton">Edit</button><button type="button" class="dangerButton">Delete</button></div></td></tr><tr><td class="tableCell">Sam Carter</td><td class="tableCell capitalize">inactive</td><td class="tableCellActions"><div class="tableRowActions"><button type="button" class="secondaryButton">Edit</button><button type="button" class="dangerButton">Delete</button></div></td></tr></tbody></table></div></div></section>"`;

exports[`PeoplePageContent renders the empty state consistently 1`] = `"<section class="section"><div class="header"><div class="headerRow"><div class="heading"><h2 class="title">People</h2><p class="subtitle">Manage individuals linked to accounts, transactions, and reimbursements.</p></div><div class="actions"><div class="buttonRow"><button type="button" class="primaryButton">Add New Person</button></div></div></div></div><div class="body"><div class="tableWrapper"><table class="table"><thead><tr><th scope="col" class="tableCell">Full Name</th><th scope="col" class="tableCell">Status</th><th scope="col" class="tableCell">Actions</th></tr></thead><tbody><tr><td colSpan="3" class="tableCell"><div class="emptyState">No people found yet.</div></td></tr></tbody></table></div></div></section>"`;
